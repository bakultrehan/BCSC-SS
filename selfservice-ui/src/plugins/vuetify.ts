import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
