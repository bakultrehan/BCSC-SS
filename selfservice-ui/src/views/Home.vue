<template>
  <div class="home">
    <HelloWorld msg="Hello world" />
  </div>
</template>

<script>
// @ is an alias to /src
import { Component, Vue } from 'vue-property-decorator';
import HelloWorld from '@/components/HelloWorld.vue';

@Component({
  components: {
    HelloWorld
  }
})
// Home goes here
export default class Home extends Vue {}
</script>
