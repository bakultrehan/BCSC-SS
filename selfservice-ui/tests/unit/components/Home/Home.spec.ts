import { shallowMount, createLocalVue } from '@vue/test-utils';
import Home from '@/components/Home/Home.vue';
import vuetify from 'vuetify';

describe('Home.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(Home, {
      localVue,
      mocks: { $t: jest.fn(() => {}) } // tslint:disable-line
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
