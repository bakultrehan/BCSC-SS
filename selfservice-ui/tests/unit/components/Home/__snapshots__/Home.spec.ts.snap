// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders props when passed 1`] = `
<v-container-stub
  fluid="true"
  tag="div"
>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="12"
      tag="div"
    >
      <h1
        class="text-center main-headline"
      >
        
      </h1>
    </v-col-stub>
     
    <v-col-stub
      class="home-img"
      cols="12"
      md="6"
      tag="div"
    />
     
    <v-col-stub
      class="text-left"
      cols="12"
      md="6"
      tag="div"
    >
      <v-card-stub
        flat="true"
        loaderheight="4"
        tag="div"
      >
        <h2
          class="second-headline"
        >
          
        </h2>
         
        <div
          class="second-sub-content"
        />
         
        <div
          class="second-sub-content"
        />
         
        <div
          class="second-sub-content"
        />
         
        <h3
          class="third-h3"
        >
          
        </h3>
         
        <div
          class="subcontent3"
        />
         
        <v-card-actions-stub
          class="btn-actions"
        >
          <button-stub
            class="btn-all"
            color="#fcba19 !important"
            name="button"
            type="button"
          >
            
          </button-stub>
           
          <button-stub
            class="btn-all getting-start-btn"
            color=""
            name="button"
            type="button"
          >
            
          </button-stub>
        </v-card-actions-stub>
      </v-card-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
